src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -256,19 +256,11 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
             HttpRequest request)
         {
             var result = new List<MediaTypeSegmentWithQuality>();
-            var parsedHeaders = request.GetTypedHeaders().Accept;
-            for (var i = 0; i < parsedHeaders?.Count; i++)
+            AcceptHeaderParser.ParseAcceptHeader(request.Headers[HeaderNames.Accept], result);
+            for (int i = 0; i < result.Count; i++)
             {
-                result.Add(new MediaTypeSegmentWithQuality(
-                    new StringSegment(parsedHeaders[i].ToString()),
-                    parsedHeaders[i].Quality ?? 1.0));
-            }
-
-            for (var i = 0; i < result.Count; i++)
-            {
-                if (!RespectBrowserAcceptHeader &&
-                    MediaTypeComparisons.MatchesAllTypes(result[i].MediaType) &&
-                    MediaTypeComparisons.MatchesAllSubtypes(result[i].MediaType))
+                var mediaType = new MediaType(result[i].MediaType);
```

Comment:
Since we parse these into `MediaTypeSegmentWithQuality` and then convert them to MediaType, is it worth just adding a quality property to `MediaType` and calling it good?